apps/meteor/client/views/admin/apps/AppsList.tsx

```diff
@@ -95,57 +80,65 @@ const AppsTable: FC<{
 		categories: useMemo(() => selectedCategories.map(({ label }) => label), [selectedCategories]),
 		purchaseType: useMemo(() => freePaidFilterStructure.items.find(({ checked }) => checked)?.id, [freePaidFilterStructure]),
 		sortingMethod: useMemo(() => sortFilterStructure.items.find(({ checked }) => checked)?.id, [sortFilterStructure]),
+		status: useMemo(() => statusFilterStructure.items.find(({ checked }) => checked)?.id, [statusFilterStructure]),
 	});
 
+	const isAppListReadyOrLoading =
+		appsResult.phase === AsyncStatePhase.LOADING || (appsResult.phase === AsyncStatePhase.RESOLVED && Boolean(appsResult.value.count));
+
+	const noInstalledAppsFound = appsResult.phase === AsyncStatePhase.RESOLVED && !isMarketplace && appsResult.value.total === 0;
+
+	const noMarketplaceOrInstalledAppMatches = appsResult.phase === AsyncStatePhase.RESOLVED && isMarketplace && appsResult.value.count === 0;
+
+	const noInstalledAppMatches =
+		appsResult.phase === AsyncStatePhase.RESOLVED && !isMarketplace && appsResult.value.total !== 0 && appsResult.value.count === 0;
+
+	const loadingRows = Array.from({ length: 8 }, (_, i) => <Skeleton key={i} height='x56' mbe='x8' width='100%' variant='rect' />);
+
 	return (
 		<>
-			{/* TODO Divide into two components: Filters and AppsTable */}
-			<FilterByText placeholder={t('Search_Apps')} onChange={({ text }): void => setText(text)}>
-				<RadioDropDown group={freePaidFilterStructure} onSelected={freePaidFilterOnSelected} mie='x8' />
-				<CategoryDropDown data={categories} selectedCategories={selectedCategories} onSelected={onSelected} />
-				<RadioDropDown group={sortFilterStructure} onSelected={sortFilterOnSelected} mis='x8' />
-			</FilterByText>
-			<TagList categories={categoryTagList} onClick={onSelected} />
-			{(appsResult.phase === AsyncStatePhase.LOADING ||
-				(appsResult.phase === AsyncStatePhase.RESOLVED && Boolean(appsResult.value.count))) && (
+			<AppsFilters
+				setText={setText}
+				freePaidFilterStructure={freePaidFilterStructure}
+				freePaidFilterOnSelected={freePaidFilterOnSelected}
+				categories={categories}
+				selectedCategories={selectedCategories}
+				onSelected={onSelected}
+				sortFilterStructure={sortFilterStructure}
+				sortFilterOnSelected={sortFilterOnSelected}
+				categoryTagList={categoryTagList}
+				statusFilterStructure={statusFilterStructure}
+				statusFilterOnSelected={statusFilterOnSelected}
+			/>
+
+			{isAppListReadyOrLoading && (
```

Comment:
How about breaking this in different components to reduce the complexity?

Reply:
Solved